src/analysis/exploration/biomassPrecursorCheck.m

```diff
@@ -14,31 +14,32 @@
 % May identify metabolites that are typically recycled within the
 % network such as ATP, NAD, NADPH, ACCOA.
 
-colS_biomass = find(model.c);
+colS_biomass = model.c ~= 0;
 % FIND COLUMN IN S-MATRIX THAT CORRESPONDS TO BIOMASS EQUATION
 
 % LIST ALL METABOLITES IN THE BIOMASS FUNCTION
-biomassMetabs = model.mets(model.S(:,colS_biomass)<0);
+biomassMetabs = model.mets(any(model.S(:, colS_biomass) < 0, 2));
 
 % ADD DEMAND REACTION, SET OBJECTIVE FUNCTION TO MAXIMIZE ITS PRODUCTION,
 % AND OPTIMIZE.  NOTE: A CRITICAL ASSUMPTION IS THAT THE ADDED DEMAND
 % REACTION IS APPENDED TO THE FAR RIGHT OF THE S-MATRIX.  THE CODE NEEDS TO
 % BE REVISED IF THIS IS NOT THE CASE.
-k=1;
-m=1;
+k = 1;
+m = 1;
 % ADD DEMAND REACTIONS
-[model_newDemand,addedRxns] = addDemandReaction(model,biomassMetabs);
-for i=1:length(biomassMetabs)
+[model_newDemand, addedRxns] = addDemandReaction(model, biomassMetabs);
+[missingMets, presentMets] = deal({});
+for i = 1:length(biomassMetabs)
 %     [model_newDemand,addedRxn] = addDemandReaction(model,biomassMetabs(i));   % ADD DEMAND REACTION
-    model_newDemand.c = zeros(length(model_newDemand.c),1);                     % CHANGE OBJECTIVE FUNCTION TO NEW DEMAND RXN
-    model_newDemand.c(strmatch(addedRxns{i},model_newDemand.rxns)) = 1;
+    model_newDemand.c = zeros(length(model_newDemand.c), 1);                     % CHANGE OBJECTIVE FUNCTION TO NEW DEMAND RXN

```

Comment:
Could you just replace these two lines by `changeObjective(model_newDemand,addedRxns{i})`